core/DataTable.php

```diff
@@ -1970,4 +1976,18 @@ public function offsetUnset($offset)
     {
         $this->deleteRow($offset);
     }
+
+    /**
+     * Returns all rows plus the totals row if it exists. Note: row IDs are not retained since the totals row has no associated ID.
+     *
+     * @return Row[]
+     */
+    public function getRowsWithTotalsRow()
+    {
+        $rows = array_values($this->getRows());
```

Comment:
That's actually the only reason I added array_values() here. The totals row doesn't have an ID currently. I thought about adding it but wasn't sure if that would cause some strange issues...

Reply:
I removed the method but couldn't add the ID, adding the totals row to `getRows()` breaks things since we'll try to add metadata & format the totals row after it's already been formatted.